src/ctree-element-screen/ctree-segment-container.html

```diff
@@ -212,6 +216,7 @@
             parent.removeChild(child);
           }
           if (type) {
+            this.$.component = this.getComponent();
```

Comment:
It looks like this is in the wrong place, because you're getting the component before creating it.  Also, shouldn't we only set the component in `this.$.component` after the component is created and the ID is set.  That way we don't need the `getComponent` (we can just inline that code here, where we create it).